packages/ethereum/test/utils/deploy.ts

```diff
@@ -0,0 +1,12 @@
+import { ethers } from 'hardhat'
+import { Contract, Signer } from 'ethers'
+
+export const deployContractFromFactory = async (
```

Comment:
Why don't make this function generic, e.g. `deployContractFromFactory<T>(...) => Promise<T>`? Would save some type casts